packages/core/src/components/reviews/ReviewModal/ReviewModal.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  Button,
+  type ModalProps as UIModalProps,
+  type ModalHeaderProps as UIModalHeaderProps,
+  type ModalBodyProps as UIModalBodyProps,
+  type ModalFooterProps as UIModalFooterProps,
+  useUI,
+} from '@faststore/ui'
+
+import styles from './section.module.scss'
+
+import dynamic from 'next/dynamic'
+
+const UIModal = dynamic<UIModalProps>(
+  () =>
+    import(/* webpackChunkName: "UIModal" */ '@faststore/ui').then(
+      (mod) => mod.Modal
+    ),
+  { ssr: false }
+)
+
+const UIModalHeader = dynamic<UIModalHeaderProps>(
+  () =>
+    import(/* webpackChunkName: "UIModalHeader" */ '@faststore/ui').then(
+      (mod) => mod.ModalHeader
+    ),
+  { ssr: false }
+)
+
+const UIModalBody = dynamic<UIModalBodyProps>(
+  () =>
+    import(/* webpackChunkName: "UIModalBody" */ '@faststore/ui').then(
+      (mod) => mod.ModalBody
+    ),
+  { ssr: false }
+)
+
+const UIModalFooter = dynamic<UIModalFooterProps>(
+  () =>
+    import(/* webpackChunkName: "UIModalFooter" */ '@faststore/ui').then(
+      (mod) => mod.ModalFooter
+    ),
+  { ssr: false }
+)
+
+interface ReviewModalProps {
+  /**
+   * The review modal's title.
+   */
+  title?: string
+  /**
+   * Close button aria-label.
+   */
+  closeButtonAriaLabel?: string
+  /**
+   * Cancel button label.
+   */
+  cancelButtonLabel: string
+  /**
+   * Submit button label.
+   */
+  submitButtonLabel: string
+}
+
+function ReviewModal({
+  title = 'Add a review',
+  closeButtonAriaLabel = 'Close Review Modal',
+  cancelButtonLabel = 'Cancel',
+  submitButtonLabel = 'Submit',
+}: ReviewModalProps) {
+  const { closeReviewModal } = useUI()
+
+  const handleSubmit = async () => {
+    // TODO: send review
+  }
+
+  function handleOnClose() {
+    closeReviewModal()
+  }
+
+  return (
+    <UIModal
+      data-fs-review-modal
+      aria-label="Review modal"
+      title={title}
+      onTransitionEnd={(_, fade) => fade === 'out' && handleOnClose()}
+      overlayProps={{
+        className: `section ${styles.section} section-review-modal`,
+      }}
+    >
+      {({ fade, fadeOut, fadeIn }) => (
```

Comment:
```suggestion
      {({ fadeOut }) => (
```
todo: remove it since we are not using. Or are we going to use it later?